autoprof/fit/base.py

```diff
@@ -66,13 +67,13 @@ def __init__(self, model, initial_state = None, relative_tolerance = 1e-3, **kwa
         self.loss_history = []
         self.message = ""
 
-    def fit(self):
+    def fit(self) -> Union[None,self]:
```

Comment:
The fitter always returns self, which means it is returning an instance of `BaseOptimizer`